### ISSUE TITLE
upgrade from 2.15.2 to 2.16.0 lost tags

### ISSUE THREAD

**Issue by bbrendon** [monicahq/monica#3419]
After upgrading from 2.15.2 to 2.16.0 all of the tags were removed from contacts. It seems like the system still knows about the tag. The Contact list doesn't show the tag names any longer until I add a tag to a contact.  

I didn't have many tagged contacts so it's not a big deal but I have a relatively new install. 

During the upgrade I looked at the log and didn't see anything wrong. I have since restarted so the log is gone. I created a backup before upgrading but I'm not very interested in restoring it.

I'm using:
mysql-community-server               5.7.27-1debian10

**Comment by asbiin** [comment]
That's ... unforunate.
We will look at this. 
@djaiss ?

**Comment by jatls** [comment]
I had the same issue with this upgrade. I did restore the contact_tag table from my backup and the tags now appear.

What may be another interesting clue to this bug is that in my case, the first tag for my first 2 contacts were not removed after the upgrade.

**Comment by yammesicka** [comment]
Had the same issue, luckily for me I had backups because I'm using Google Cloud SQL.

**Comment by asbiin** [comment]
I've run some tests, and I did not find any deleted tags during the upgrade.

**Comment by mechanarchy** [comment]
I found this issue too - on a self-hosted instance I upgraded from 2.15.2 to 2.16.0, then immediately to 2.17.0. At the end, the tags existed, but no contacts had been linked with any of them.

I ended up restoring a backup for 2.15.2 and then upgrading only to 2.16.0 and this worked for me (ie tags were linked correctly). I'll probably skip 2.17.0 if possible because of a couple of other problems (eg #3743).
